pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -567,6 +582,183 @@ func (v *vpnSeedServer) Deploy(ctx context.Context) error {
 		return err
 	}
 
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, envoyFilter, func() error {
```

Comment:
See #4227, maybe you can go with `GetAndCreateOrMergePatch` for this new code right away.

Reply:
Used it for the new code and adapted the tests accordingly.